### ISSUE TITLE
Add remaining type hints

### ISSUE THREAD

**Issue by isabelcosta** [anitab-org/mentorship-backend#505]
## Description
As a developer,
I need type hints to be added to the code,
so that I can improve developers' experience.

Some type hints were added in this PR #354 , in this issue you should add all the remaining type hints you may find.

helpful resources:
- [PEP 484 -- Type Hints](https://www.python.org/dev/peps/pep-0484/)
- [PR #354 that added type hints to the project](https://github.com/anitab-org/mentorship-backend/pull/354)

### Update [Required]
- [ ] Add remaining type hints missing

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
4 hours

**Comment by AjayThakur12** [comment]
@isabelcosta I can work on a specific module or for complete project

**Comment by isabelcosta** [comment]
I actually prefer on a specific module for eas of review, which module do you wish to add type hints to? @AjayThakur12 Can you create an issue for the specific module you wish to solve and add a reference to this issue, for you to focus your work on that module?

**Comment by isabelcosta** [comment]
> @isabelcosta Thanks for mentioning this issue to me. I am interested to work on it and will take a look at it. Could you assign this to me then? And if I need any help, I hope I can reach out on this thread for any assistance? Thank You :)

Yes I can assign to you :) and yes, of course, you can reach out @sankha555

**Comment by isabelcosta** [comment]
Can you comment here, so I can assign you here?

**Comment by sankha555** [comment]
@isabelcosta You can assign me the issue. Hoping to resolve this issue as quick as posssible :)

**Comment by isabelcosta** [comment]
Done @sankha555

**Comment by sankha555** [comment]
Hello @isabelcosta,
I have added typehints to a few modules (which I have mentioned in the above commit #569). Could you please review my work and tell me if the changes made are okay? Based on your feedback, I would resume adding the remaining typehints to the rest of the modules. 

Thanks!

**Comment by sankha555** [comment]
Hello @isabelcosta ,
First of all I deeply regret and apologise for all the inconvenience caused to you yesterday due to errors made by me in making a correct PR. I am absolutely a newbie to open-source contributions and it took me some time to figure out what's going wrong. I hope you would understand my situation. 

I request you to kindly review my PR (#572)  made from a new branch. I hope this time there haven't been any goof-ups from my side. I sincerely await your review and feedback on the task.
Thanks!

**Comment by isabelcosta** [comment]
> First of all I deeply regret and apologise for all the inconvenience caused to you yesterday due to errors made by me in making a correct PR. I am absolutely a newbie to open-source contributions and it took me some time to figure out what's going wrong. I hope you would understand my situation.

No need to apologize at all! Newbie to open source? That is exciting! I will be happy if after this issue you feel more comfortable in this space :) I can guide you with no problem if you have doubts and questions just ask. So no need to apologize, no inconvenience was caused 🤗 

> I request you to kindly review my PR (#572) made from a new branch. I hope this time there haven't been any goof-ups from my side. I sincerely await your review and feedback on the task.

I will do as soon as I get some time :)
Already some good 🍪 points:
- you followed the PR template
- you created a separate branch for your work
- you described very well your changes in the PR, with the modules you updated

You are in a good path for making a successful contribution 🤗 💪 @sankha555

**Comment by sankha555** [comment]
Thanks a lot for the motivation @isabelcosta 🙂! I hope to make more contributions to the codebase( without the goof-ups this time onwards 😅). Would wait for you to review my work. Meanwhile,  if you think here are more issues that I can solve, please mention me in those. I would be eager to work on them. Thanks again :)

**Comment by sankha555** [comment]
@isabelcosta, I request you to not review PR #572. I think I have discovered an issue in my code which might prove problematic. I would close this PR and make a new one. I would notify here about the same. Please review that PR and not this one(#572), whenever you may find time. Thanks!

**Comment by isabelcosta** [comment]
@sankha555 on this note, I want to ask you something. From what I saw from the PRs you sent, is thst they do the right thing. For the next PR you send, could you only commit changes for a specific module. This makes the code review a bit more easier to do, since it's less code to look at and the commits have a logical scope, per module. For example send a PR just for type ints in the utils module. Makes sense?

Another thing, is that you don't need to close and create new PRs to add or remove code from the PR. You can do that on the current PR/branch you are working on. So next time please wait before closing a PR, I can help you in what you need to do to change code on the PR. No worries about having more than one commit in your PR :)

**Comment by sankha555** [comment]
@isabelcosta I get your point. I will stick to lesser no. of changes from next time onwards. So if you say that my code was fine, shall I reopen the PR then?

**Comment by isabelcosta** [comment]
Yes, in my opinion you can open your PR for sure :)

**Comment by sankha555** [comment]
Done @isabelcosta . Another thing I wanted to know is that in future if I need to change something in my PR, which would be a better method to do so in your opinion? Should I make a new commit to the current branch or revert all commits in the branch and make a new branch altogether?

**Comment by isabelcosta** [comment]
Simply create new commits, it's the simplest way to do it, and a correct way :) @sankha555

**Comment by sankha555** [comment]
Sure @isabelcosta , I will take care of these things from the next time. So are my current PR contributions okay completely by what you reviewed? I was quite eager to know if I have made my first code contribution correctly :)

**Comment by isabelcosta** [comment]
I haven't yet make time to check it properly, and since it's so much code it's a bit time consuming to check it all. I will as soon as I have time for it. @sankha555

**Comment by sankha555** [comment]
Sure, @isabelcosta :)

**Comment by to-sta** [comment]
Hi, i am a frist timer and was just looking for issues to work on. As it is still open and available i am intrested in tackling this issue :)
